File: ios/RebelChat/Classes/Helper/Easing.swift
//
//  Easing.swift
//  RebelChat
//
//  Created by Émile Bélair on 2016-03-08.
//  Copyright © 2016 Mirego. All rights reserved.
//

import Foundation

func easeIn(x: CGFloat) -> CGFloat
{
    return max(0, min(1, x * x))
}

func easeOut(x: CGFloat) -> CGFloat
{
    return max(0, min(1, 1 - (1 - x) * (1 - x)))
}

func easeInOut(x: CGFloat) -> CGFloat
{
    return max(0, min(1, x < 0.5 ? (2 * x * x) : -1 + (4 - 2 * x) * x))
}

File: android/RebelChat/app/src/main/java/com/mirego/rebelchat/utilities/Easing.java
package com.mirego.rebelchat.utilities;

public class Easing {

    public static float easeIn(float x) {
        return Math.max(0, Math.min(1, x * x));
    }

    public static float easeOut(float x) {
        return Math.max(0, Math.min(1, 1 - (1 - x) * (1 - x)));
    }

    public static float easeInOut(float x) {
        return Math.max(0, Math.min(1, x < 0.5 ? (2 * x * x) : -1 + (4 - 2 * x) * x));
    }
}
